### ISSUE TITLE
xwayland apps not registering clicks on left side of the screen/workspace

### ISSUE THREAD

**Issue by mxaddict** [hyprwm/Hyprland#6224]
### Hyprland Version

<details> <summary>System/Version info</summary>

```sh
Hyprland, built from branch  at commit cba1ade848feac44b2eda677503900639581c3f4  (props: bump version to 0.40.0).
Date: Sat May 4 15:42:32 2024
Tag: v0.40.0, commits: 4606

flags: (if any)


System Information:
System name: Linux
Node name: mxaddict-pc
Release: 6.9.1-arch1-2
Version: #1 SMP PREEMPT_DYNAMIC Wed, 22 May 2024 13:47:07 +0000


GPU information:
03:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Navi 31 [Radeon RX 7900 XT/7900 XTX/7900M] [1002:744c] (rev c8) (prog-if 00 [VGA controller])
18:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Raphael [1002:164e] (rev c1) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

xwayland apps randomly start being only clickable on left side of my monitor as if the coordinates for the app are moved half offscreen.

Happens to a bunch of apps, namely:
Steam,
Spotify-Launcher
Lutris games
Anything running in proton
And some other random xwayland qt apps

### How to reproduce

Easiest way to replicate is to start a game in steam via proton and close it after about 2-3 restarts it will start happening.

**Comment by mxaddict** [comment]
I will test a git build and see if it's fixed, thanks for the pointer

**Comment by mxaddict** [comment]
I running build from the https://github.com/hyprwm/Hyprland/pull/6086 PR and no issues so far, will report back once I've used it a bit longer.

**Comment by erlonbie** [comment]
@mxaddict have you managed to fix it? I'm on 0.41.1 with the same bug. I have 3 monitors, 2 horizontal (one on top of the other) and 1 vertical at left side of both.  It happen on chrome, postman and maybe other qt apps. Gtk ones seems to be fine. Also, it only happens on 1 of my monitors (the bottom one)

**Comment by mxaddict** [comment]
No, this still happens every once in a while, but restarting fixes it
temporarily.


Cheers,
Barry Deeney

**Comment by dotaxis** [comment]
I think I'm experiencing this on Steam.

It works if I move it to my right monitor but does not work on my left monitor. I only discovered that it worked on my right monitor after finding this issue.

My monitors are configured as so:
```
monitor=DP-2, 2560x1440@165, 0x0, 1
monitor=DP-1, 2560x1440@165, 2560x0, 1
```

I use `split-monitor-workspaces` so technically I think the workspaces on the left monitor are numbered 11-20.

I'll test further after work now that I know it works on one monitor but not the other.

**Comment by veRYANgry** [comment]
Regressing to the pull before https://github.com/hyprwm/Hyprland/pull/6086 at https://github.com/hyprwm/Hyprland/pull/6342 fixes the issue for me.

**Comment by DrShadow34** [comment]
Also have that issue time to time, without any obvious pattern in overwatch. It very annoying, cause you need to restart xwayland and that crashes hyprland for some reason. I have only one monitor. Also, draging game window to the left half of the screen help. But making window fullscreen shift cursor further away for some reason.

**Comment by romek-codes** [comment]
I'm also having the same issue, from time to time on my left vertical monitor, it seems to be happening most often on Discord apps, doesn't matter if it's webcord or vesktop, in both of these apps it occurs often without a clear pattern as to why. If i move it to my right monitor i can click everything normally.

**Comment by PaideiaDilemma** [comment]
Potential fix: https://github.com/hyprwm/Hyprland/pull/9179
